prow/githubeventserver/options.go

```diff
@@ -20,48 +20,22 @@ import (
 	"flag"
 	"fmt"
 	"strings"
-
-	"github.com/sirupsen/logrus"
 )
 
 // Options holds the endpoint and port information that can be used
 // to create a new github event server
 type Options struct {
-	// HmacTokenGenerator is a function that holds a hmac token that will be used
-	// in a github event server
-	HmacTokenGenerator func() []byte
-
-	// Metrics will be used to expose prometheus metrics from the
-	// github event server operations.
-	Metrics *Metrics
-
-	// Logger is the logger that the github event server will use.
-	Logger *logrus.Entry
-
 	// endpoint is the main url path that the github event server will be served.
 	endpoint string
 	// port will be used to start an http server to listen to.
 	port int
 }
 
-// DefaultAndValidate validates the option's values and defaults them if they are empty.
-func (o *Options) DefaultAndValidate() error {
+// Validate validates the option's values and defaults them if they are empty.
+func (o *Options) Validate() error {
 	if !strings.HasPrefix(o.endpoint, "/") {
 		return fmt.Errorf("endpoint %s is not a valid url path", o.endpoint)
 	}
-
-	if o.HmacTokenGenerator == nil {
-		o.HmacTokenGenerator = func() []byte { return []byte("") }
-	}
-
-	if o.Metrics == nil {
```

Comment:
If you were already doing this, then you do not need to change anything, right? Passing a `NewMetrics()` manually does nothing on top of the default.

Reply:
updated.